src/ark/utils/deepcell_service_utils.py

```diff
@@ -99,85 +176,41 @@ def create_deepcell_output(deepcell_input_dir, deepcell_output_dir, fovs=None,
 
     print(f'Processing tiffs in {len(fov_groups)} batches...')
 
-    # yes this is function, don't worry about it
-    # long story short, too many args to pass if function not in local scope
-    # i.e easier to map fov_groups
-    def _zip_run_extract(fov_group, group_index):
-        # define the location of the zip file for our fovs
-        zip_path = os.path.join(deepcell_input_dir, f'fovs_batch_{group_index + 1}.zip')
-
-        if os.path.isfile(zip_path):
-            warnings.warn(f'{zip_path} will be overwritten')
-
-        # write all files to the zip file
-        print('Zipping preprocessed tiff files.')
-
-        def zip_write(zip_path):
-            with ZipFile(zip_path, 'w', compression=ZIP_DEFLATED) as zipObj:
-                for fov in fov_group:
-                    # file has .tiff extension
-                    basename = fov + '.tiff'
-                    filename = os.path.join(deepcell_input_dir, basename)
-                    zipObj.write(filename, basename)
-
-        zip_write(zip_path)
-
-        # pass the zip file to deepcell.org
-        print('Uploading files to DeepCell server.')
-        status = run_deepcell_direct(
-            zip_path, deepcell_output_dir, host, job_type, scale, timeout
-        )
-
-        # ensure execution is halted if run_deepcell_direct returned non-zero exit code
-        if status != 0:
-            print("The following FOVs could not be processed: %s" % ','.join(fov_group))
-            return
+    for batch_num, fov_group in enumerate(fov_groups):
```

Comment:
In `zip_input_files` and `extract_deep_cell_response` we add 1 for the `batch_num`. We can start the enumeration at `1` with `enumerate(Iterable, start=1)` and remove that op to make it easier to read overall.

Reply:
oh this is v helpful thanks!